src/components/ResourceDistribution.vue

```diff
@@ -0,0 +1,267 @@
+<template>
+  <div>
+    <h1 class="text-center">
+      Current leadership distribution across brokers per topic
+    </h1>
+
+    <div class="col-5 mx-auto">
+      <div class="row">
+        <div class="col-8 mx-auto">
+          <input
+            type="text"
+            class="form-control"
+            v-model="filter"
+            placeholder="filter topics.."
+            :disabled="stacked"
+          />
+        </div>
+        <div class="col-4 mx-auto">
+          <div class="form-check">
+            <input
+              v-model="stacked"
+              type="checkbox"
+              class="form-check-input"
+              id="stackedCheckBox"
+            />
+            <label class="form-check-label" for="stackedCheckBox">
+              Stacked view
+            </label>
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <div class="alert alert-danger" role="alert" v-if="error">
+      {{ error }}
+    </div>
+
+    <div class="row" v-bind:class="{ active: loaded }" v-show="stacked">
+      <div class="col-8 mx-auto">
+        <div class="card mx-4 mt-4">
+          <div class="card-inner">
+            <LineChart
+              :chart-data="formatStackedData"
+              :options="getStackedOptions"
+            ></LineChart>
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <div
+      class="row mx-auto"
+      v-bind:class="{ active: loaded }"
+      v-show="!stacked"
+    >
+      <div
+        class="col-4"
+        v-for="item in cachedKccData"
+        v-bind:key="item[0]"
+        v-show="item[0].toLowerCase().includes(filter.toLowerCase())"
+      >
+        <div class="card mt-4">
+          <div class="card-inner">
+            <LineChart
+              :chart-data="formatItemData(item, 'leaders')"
+              :options="formatItemOptions(item)"
+            ></LineChart>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+
+<script>
+import LineChart from '@/components/LineChart.vue'
+const backgroundColors = ['#ffffd9', '#edf8b1', '#c7e9b4', '#7fcdbb', '#41b6c4', '#1d91c0', '#225ea8', '#253494', '#081d58']
+
+class Topic {
+  constructor () {
+    this.leaders = {}
+    this.replicas = {}
+    this.replicationFactor = 0
+  }
+
+  addLeader (item) {
+    let broker = item.leader
+    if (this.leaders[broker] === undefined) {
+      this.leaders[broker] = 1
+      return
+    }
+    this.leaders[broker]++
+  }
+
+  addReplicas (item) {
+    let replicas = Object.values(item.followers)
+    replicas.push(item.leader)
+
+    replicas.map(follower => {
+      if (this.replicas[follower] === undefined) {
+        this.replicas[follower] = 1
+        return
+      }
+
+      this.replicas[follower]++
+    })
+  }
+}
+
+export default {
+  name: 'ResourceDistribution',
```

Comment:
should we add `Chart` suffix for all chart components we are gonna develop ?

ex: `ResourceDistributionChart`